src/Form.php

```diff
@@ -0,0 +1,378 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+use Yiisoft\Strings\Inflector;
+
+abstract class Form implements FormInterface
+{
+    private array $attributes = [];
+    private array $attributesLabels = [];
+    private array $errors = [];
+
+    public function __construct()
+    {
+        $this->attributes = $this->attributes();
+    }
+
+    /**
+     * Adds a new error to the specified attribute.
+     *
+     * @param string $attribute attribute name
+     * @param string $error new error message
+     */
+    public function addError(string $attribute, string $error): void
+    {
+        $this->errors[$attribute][] = $error;
+    }
+
+    /**
+     * Adds a list of errors.
+     *
+     * @param array $items a list of errors. The array keys must be attribute names.
+     * The array values should be error messages. If an attribute has multiple errors, these errors must be given in
+     * terms of an array.
+     *
+     * You may use the result of {@see getErrors()} as the value for this parameter.
+     */
+    public function addErrors(array $items): void
+    {
+        foreach ($items as $attribute => $errors) {
+            if (\is_array($errors)) {
+                foreach ($errors as $error) {
+                    $this->addError($attribute, $error);
+                }
+            } else {
+                $this->addError($attribute, $errors);
+            }
+        }
+    }
+
+    /**
+     * Returns the list of attribute names.
+     *
+     * By default, this method returns all private non-static properties of the class.
+     *
+     * @return array list of attribute names.
+     */
+    public function attributes(): array
+    {
+        $class = new \ReflectionClass($this);
```

Comment:
I think you need to do this in constructor instead of public method

Reply:
done.